evm/chains/sharding/mainchain_handler/vmc_utils.py

```diff
@@ -0,0 +1,135 @@
+
+WITHDRAW_HASH = keccak("withdraw")
+
+
+def mk_validation_code(address):
+    """
+    validation_code = '''
+~calldatacopy(0, 0, 128)
+~call(3000, 1, 0, 0, 128, 0, 32)
+return(~mload(0) == {})
+    '''.format(utils.checksum_encode(address))
+    return serpent.compile(validation_code)
+    """
+    # The precompiled bytecode of the validation code which
+    # verifies EC signatures
+    validation_code_bytecode = b"a\x009\x80a\x00\x0e`\x009a\x00GV`\x80`\x00`\x007` "
+    validation_code_bytecode += b"`\x00`\x80`\x00`\x00`\x01a\x0b\xb8\xf1Ps"
+    validation_code_bytecode += address
+    validation_code_bytecode += b"`\x00Q\x14` R` ` \xf3[`\x00\xf3"
+    return validation_code_bytecode
+
+
+def sign(message, privkey):
+    """@privkey: Key type
+    """
+    signature = keys.ecdsa_sign(message, privkey)
+    v, r, s = signature.vrs
+    v += 27
+    signature_bytes = b''.join([item.to_bytes(32, 'big') for item in (v, r, s)])
+    return signature_bytes
+
+
+def get_vmc_json():
+    mydir = os.path.dirname(__file__)
+    vmc_path = os.path.join(mydir, '../contracts/validator_manager.json')
+    vmc_json_str = open(vmc_path).read()
+    return json.loads(vmc_json_str)
+
+
+def get_contract_address_from_contract_tx(transaction):
+    return pipe(
+        transaction.sender,
+        to_canonical_address,
+        functools.partial(generate_contract_address, nonce=0),
+    )
+
+
+def create_vmc_tx(TransactionClass, gasprice=GASPRICE):
+    vmc_json = get_vmc_json()
+    vmc_bytecode = decode_hex(vmc_json['bytecode'])
+    v = 27
+    r = 1000000000000000000000000000000000000000000000000000000000000000000000000000
+    s = 1000000000000000000000000000000000000000000000000000000000000000000000000000
+    return TransactionClass(0, gasprice, 3000000, b'', 0, vmc_bytecode, v, r, s)
+
+
+def create_transaction_from_hex(raw_transaction_hex, TransactionClass):
+    return pipe(
+        raw_transaction_hex,
+        decode_hex,
+        functools.partial(rlp.decode, sedes=TransactionClass),
+    )
+
+
+def create_sighasher_tx(TransactionClass):
+    sighasher_tx_hex = "0xf9016d808506fc23ac0083026a508080b9015a6101488061000e6000396101565660007f01000000000000000000000000000000000000000000000000000000000000006000350460f8811215610038576001915061003f565b60f6810391505b508060005b368312156100c8577f01000000000000000000000000000000000000000000000000000000000000008335048391506080811215610087576001840193506100c2565b60b881121561009d57607f8103840193506100c1565b60c08112156100c05760b68103600185013560b783036020035260005101840193505b5b5b50610044565b81810360388112156100f4578060c00160005380836001378060010160002060e052602060e0f3610143565b61010081121561010557600161011b565b6201000081121561011757600261011a565b60035b5b8160005280601f038160f701815382856020378282600101018120610140526020610140f350505b505050505b6000f31b2d4f"  # noqa: E501
+    return create_transaction_from_hex(sighasher_tx_hex, TransactionClass)
+
+
+def create_viper_rlp_decoder_tx(TransactionClass):
```

Comment:
Since this is only used in testing, what do you think about moving it into the appropriate module under `tests/`